apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/contacts/ContactsView.kt

```diff
@@ -0,0 +1,385 @@
+                .distinctUntilChanged()
+                .collect {
+                    if (it.isNotEmpty()) {
+                        focusRequester.requestFocus()
+                    } else if (listState.layoutInfo.totalItemsCount > 0) {
+                        listState.scrollToItem(0)
+                    }
+                }
+        }
+    }
+}
+
+@Composable
+private fun ToggleFilterButton() {
+    val pref = remember { ChatController.appPrefs.showUnreadAndFavorites }
+    IconButton(onClick = { pref.set(!pref.get()) }) {
+        val sp16 = with(LocalDensity.current) { 16.sp.toDp() }
+        Icon(
+            painterResource(MR.images.ic_filter_list),
+            null,
+            tint = if (pref.state.value) MaterialTheme.colors.background else MaterialTheme.colors.secondary,
+            modifier = Modifier
+                .padding(3.dp)
+                .background(color = if (pref.state.value) MaterialTheme.colors.primary else Color.Unspecified, shape = RoundedCornerShape(50))
+                .border(width = 1.dp, color = if (pref.state.value) MaterialTheme.colors.primary else Color.Unspecified, shape = RoundedCornerShape(50))
+                .padding(3.dp)
+                .size(sp16)
+        )
+    }
+}
+
+private var lazyListState = 0 to 0
+
+@Composable
+private fun ContactsList(
+    contactActions: @Composable () -> Unit,
+    contactTypes: List<ContactType>,
+    contactListTitle: String ? = null
+) {
+    val listState = rememberLazyListState(lazyListState.first, lazyListState.second)
+    val searchText = rememberSaveable(stateSaver = TextFieldValue.Saver) { mutableStateOf(
+        TextFieldValue("")
+    ) }
+
+    var searchFocused by remember { mutableStateOf(false) }
+
+    DisposableEffect(Unit) {
+        onDispose {
+            lazyListState =
+                listState.firstVisibleItemIndex to listState.firstVisibleItemScrollOffset
+        }
+    }
+    val showUnreadAndFavorites =
+        remember { ChatController.appPrefs.showUnreadAndFavorites.state }.value
+
+    val allChats by remember(chatModel.chats, contactTypes) {
+        derivedStateOf { contactChats(chatModel.chats, contactTypes) }
+    }
+
+    val filteredContactChats = filteredContactChats(
+        showUnreadAndFavorites = showUnreadAndFavorites,
+        searchText = searchText.value.text,
+        contactChats = allChats
+    )
+
+    LazyColumn(
+        Modifier.fillMaxWidth(),
+        listState
+    ) {
+        item {
+            SectionView {
+                Divider()
+                ContactsSearchBar(
+                    listState = listState,
+                    searchText = searchText,
+                    focused = searchFocused,
+                    onFocusChanged = {
+                        searchFocused = it
+                    }
+                )
+                Divider()
+            }
+
+            if (!searchFocused) {
+                contactActions()
+            }
```

Comment:
I think these action buttons (Add/Scan/Create group) should have a bit more padding from search, would look better